test/mocha-test.js

```diff
@@ -0,0 +1,89 @@
+const mongoose = require('mongoose');
+const ProductFeatures = require('../database/database.js');
+const { fakeDataGenerator } = require('../database/seed-data/fakeDataGenerator.js');
+
+// require dev-dependencies for testing
+const assert = require('assert');
+const chai = require('chai'), chaiHttp = require('chai-http'), expect = require('chai').expect;
+chai.use(chaiHttp);
+const request = require('request');
+
+
+describe('Database seeding', () => {
+  describe('Fake Data Generator', () => {
+    it('should generate 100 records when 100 records are requested', () => {
+      let fakeTestData = fakeDataGenerator(100, 0);
+      assert.equal(fakeTestData.length, 100);
+    });
+  });
+
+  describe('Seeding Script', () => {
+    it('throws an error if no parameters are passed into fakeDataGenerator()', (done) => {
+      assert.throws(() => fakeDataGenerator(), Error, 'Must include recordsToCreate and startingRecordID params')
+      done();
+    });
+
+    // open mongoose connection for database saving
+    mongoose.connect('mongodb://localhost/fec_product_features');
+    mongoose.connection.on('error', console.error.bind(console, 'connection error:'));
+
+    // before running testing, delete test document record
+    beforeEach((done) => {
+      ProductFeatures.deleteOne({ productId: 1 }, (err) => {
+        done();
+      });
+    });
+
+    it('creates and saves record(s) in the database successfully', (done) => {
+      let fakeTestData = fakeDataGenerator(1, 1);
+      let fakeProductFeatures = new ProductFeatures(fakeTestData[0]);
+      fakeProductFeatures.save()
+        .then(() => {
+          assert(!fakeProductFeatures.isNew);
+        })
+        .finally(() => {
+          mongoose.connection.close();
+          done();
+        });
```

Comment:
You could possibly use the mocha hook `after()` instead of closing the mongoose connection in this block. So that if you want to add more tests below you can keep the order that you have without having to move the mongoose.connection.close around.

Reply:
Thanks for pointing that out. I was experiencing issues, but maybe because I was using the afterEach() hook. I will refactor and try that! Thanks so much.